src/push/mod.rs

```diff
@@ -48,26 +48,46 @@ impl PushToken {
     }
 }
 
+impl AsRef<str> for FcmToken {
+    fn as_ref(&self) -> &str {
+        self.0.as_str()
+    }
+}
+
 /// Payload sent to end device inside the push notification.
 #[derive(Debug, Serialize)]
-struct ThreemaPayload<'a> {
+pub(super) struct ThreemaPayload<'a> {
     /// Session id (public key of the initiator)
-    wcs: &'a str,
+    #[serde(rename = "wcs")]
+    session_id: &'a str,
     /// Affiliation id
-    wca: Option<&'a str>,
+    #[serde(rename = "wca")]
+    affiliation_id: Option<&'a str>,
     /// Timestamp
-    wct: i64,
+    #[serde(rename = "wct")]
+    #[serde(serialize_with = "serialize_as_str")]
+    timestamp: i64,
     /// Version
-    wcv: u16,
+    #[serde(rename = "wcv")]
+    #[serde(serialize_with = "serialize_as_str")]
+    version: u16,
```

Comment:
I'm pretty sure the `serialize_as_str` part is wrong (in the sense of a backwards-incompatible change), isn't it?

https://github.com/threema-ch/push-relay/?tab=readme-ov-file#push-payload

Reply:
It's not wrong for FCM, so I had to create separate variants for FCM and everything else